validator/tasks/task_prep.py

```diff
@@ -36,14 +38,35 @@ async def upload_json_to_minio(file_path: str, bucket_name: str, object_name: st
         return False
 
 
-def train_test_split(dataset_name: str, test_size: float = None) -> DatasetDict:
+async def load_dataset_from_s3(dataset_url: str) -> Dataset | DatasetDict:
+    """Load a dataset from S3 storage."""
+    try:
+        local_file_path = await download_s3_file(dataset_url)
+        # Create temp directory and move file there
+        temp_dir = tempfile.mkdtemp()
+        filename = os.path.basename(local_file_path)
+        new_path = os.path.join(temp_dir, filename)
+        os.rename(local_file_path, new_path)
+        dataset = load_dataset(temp_dir)
+        # Cleanup
+        os.remove(new_path)  # needed to remove the directory
+        os.rmdir(temp_dir)
+        return dataset
+    except Exception as e:
+        logger.exception(f"Failed to load dataset from S3: {e}")
+        raise e
```

Comment:
I suggest using a context manager here to automatically cleanup the files instead of using os.remove and os.rmdir:

```suggestion
async def load_dataset_from_s3(dataset_url: str) -> Dataset | DatasetDict:
    """Load a dataset from S3 storage."""
    try:
          with tempfile.TemporaryDirectory() as temp_dir:
            local_file_path = await download_s3_file(dataset_url)
           
            filename = os.path.basename(local_file_path)
            new_path = os.path.join(temp_dir, filename)
            
            os.rename(local_file_path, new_path)
            
            dataset = load_dataset(temp_dir)
            
            return dataset
    except Exception as e:
        logger.exception(f"Failed to load dataset from S3: {e}")
        raise e
```